flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BufferWritingResultPartition.java

```diff
@@ -363,7 +363,7 @@ private void finishBroadcastBufferBuilder() {
         if (broadcastBufferBuilder != null) {
             numBytesOut.inc(broadcastBufferBuilder.finish() * numSubpartitions);
             numBuffersOut.inc(numSubpartitions);
-            broadcastBufferBuilder.recycle();
+            broadcastBufferBuilder.close();
```

Comment:
You are modifying a code that you have introduced in the same PR (increasing PR size). It would be better to have reversed order of the commits.